ADAL/src/request/ADWebAuthResponse.m

```diff
@@ -112,19 +112,24 @@ - (void)handleResponse:(ADWebResponse *)webResponse
         case 400:
         case 401:
         {
-            if(!_request.handledPkeyAuthChallenge)
+
+            NSString* wwwAuthValue = [webResponse.headers valueForKey:wwwAuthenticateHeader];
+            if(![NSString adIsStringNilOrBlank:wwwAuthValue] && [wwwAuthValue adContainsString:pKeyAuthName])
             {
-                NSString* wwwAuthValue = [webResponse.headers valueForKey:wwwAuthenticateHeader];
-                if(![NSString adIsStringNilOrBlank:wwwAuthValue] && [wwwAuthValue adContainsString:pKeyAuthName])
-                {
-                    [self handlePKeyAuthChallenge:wwwAuthValue
-                                       completion:completionBlock];
-                    return;
-                }
+                [self handlePKeyAuthChallenge:wwwAuthValue
+                                   completion:completionBlock];
+                return;
             }
             
-            [self handleJSONResponse:webResponse completionBlock:completionBlock];
-            break;
+            if(_request.acceptOnlyOKResponse && webResponse.statusCode != 200)
```

Comment:
A case statement where we're checking a condition from a previous case statement? I think we're stretched the limit of having a sensible switch/case statement here, but I'm somewhat at a loss as to what would make this better/easier to follow.